pandas/_libs/tslibs/strptime.pyx

```diff
@@ -0,0 +1,627 @@
+                    week_of_year_start = 6
+                else:
+                    # W starts week on Monday.
+                    week_of_year_start = 0
+            elif parse_code == 17:
+                # Since -1 is default value only need to worry about setting tz
+                # if it can be something other than -1.
+                found_zone = found_dict['Z'].lower()
+                for value, tz_values in enumerate(locale_time.timezone):
+                    if found_zone in tz_values:
+                        # Deal w/ bad locale setup where timezone names are the
+                        # same and yet time.daylight is true; too ambiguous to
+                        # be able to tell what timezone has daylight savings
+                        if (time.tzname[0] == time.tzname[1] and
+                            time.daylight and found_zone not in (
+                                "utc", "gmt")):
+                            break
+                        else:
+                            tz = value
+                            break
+        # If we know the wk of the year and what day of that wk, we can figure
+        # out the Julian day of the year.
+        if julian == -1 and week_of_year != -1 and weekday != -1:
+            week_starts_Mon = True if week_of_year_start == 0 else False
+            julian = _calc_julian_from_U_or_W(year, week_of_year, weekday,
+                                              week_starts_Mon)
+        # Cannot pre-calculate datetime_date() since can change in Julian
+        # calculation and thus could have different value for the day of the wk
+        # calculation.
+        try:
+            if julian == -1:
+                # Need to add 1 to result since first day of the year is 1, not
+                # 0.
+                julian = datetime_date(year, month, day).toordinal() - \
+                    datetime_date(year, 1, 1).toordinal() + 1
+            else: # Assume that if they bothered to include Julian day it will
+                # be accurate.
+                datetime_result = datetime_date.fromordinal(
+                    (julian - 1) + datetime_date(year, 1, 1).toordinal())
+                year = datetime_result.year
+                month = datetime_result.month
+                day = datetime_result.day
+        except ValueError:
+            if is_coerce:
+                iresult[i] = NPY_NAT
+                continue
+            raise
+        if weekday == -1:
+            weekday = datetime_date(year, month, day).weekday()
+
+        dts.year = year
+        dts.month = month
+        dts.day = day
+        dts.hour = hour
+        dts.min = minute
+        dts.sec = second
+        dts.us = us
+        dts.ps = ns * 1000
+
+        iresult[i] = pandas_datetimestruct_to_datetime(PANDAS_FR_ns, &dts)
+        if check_dts_bounds(&dts):
+            if is_coerce:
+                iresult[i] = NPY_NAT
+                continue
+            else:
```

Comment:
maybe worth it having a tiny extension util.pyx that houses things  like this (and checknull for example)
something that we can import anywhere that isn't a dependency itself

Reply:
actually i bet inference can be this extension

Reply:
> inference.pyx can be this extension

? inference.pyx depends on a bunch of stuff, including `tslib`.

The solution I've used locally is to put this into a `tslibs.npy_dtime` which has no dependencies.

Reply:
ok, the name needs work though :>

Reply:
> ok, the name needs work though :>

OK, as long as it isn't another thing named `datetime`.